vertical-pod-autoscaler/enhancements/8818-in-place-only/README.md

```diff
@@ -0,0 +1,245 @@
+- Feature Gate Disabled: Verify InPlace mode is rejected when feature gate is disabled
+- Indefinite Wait for In-Progress Updates: Update is in-progress for extended period, verify no timeout/eviction occurs (unlike `InPlaceOrRecreate`)
+- Failed Update - Retry Success: Update fails initially, conditions improve, verify successful retry
+- Infeasible Resize Handling: Pod resize marked as infeasible, verify pod is deferred (not evicted)
+
+## Upgrade / Downgrade Strategy
+
+### Upgrade
+
+On upgrade to VPA 1.6.0 (tentative release version), users can opt into the new `InPlace` mode by enabling the alpha Feature Gate (which defaults to disabled) by passing `--feature-gates=InPlace=true` to the updater and admission-controller components and setting their VPA UpdateMode to use `InPlace`.
+Existing VPAs will continue to work as before.
+
+### Downgrade
+
+On downgrade of VPA from 1.6.0 (tentative release version), nothing will change. VPAs will continue to work as previously, unless, the user had enabled the feature gate. In which case downgrade could break their VPA that uses `InPlace`.
+
+## Feature Enablement and Rollback
+
+### How can this feature be enabled / disabled in a live cluster?
+
+- Feature gate name: `InPlace`
+- Components depending on the feature gate:
+    - admission-controller
+    - updater
+
+Disabling of feature gate `InPlace` will cause the following to happen:
+- admission-controller will:
+	- Reject new VPA objects being created with `InPlace` configured
+    	- A descriptive error message should be returned to the user letting them know that they are using a feature gated feature
+	- Continue to apply recommendations at pod admission time for existing VPAs configured with InPlace mode (behaving similarly to Initial mode)
+		- This ensures that when pods are deleted and recreated (e.g., by a deployment rollout or manual deletion), they receive the latest resource recommendations
+		- Only the in-place update functionality is disabled; admission-time updates remain functional
+- updater will:
+	- Skip processing VPAs with `InPlace` mode (no in-place updates or evictions will be attempted)
+	- Effectively treating these VPAs as if they were in Initial or Off mode for running pods
+
+Enabling of feature gate `InPlace` will cause the following to happen:
+- admission-controller to accept new VPA objects being created with `InPlace` configured
+- updater will attempt to perform an in-place **only** adjustment for VPAs configured with `InPlace`
+
+## Kubernetes version compatibility
+
+`InPlace` is being built assuming that it will be running on a Kubernetes version of at least 1.33 with the beta version of [KEP-1287: In-Place Update of Pod Resources](https://github.com/kubernetes/enhancements/issues/1287) enabled.
+Should these conditions not be true, the VPA shall not be able to scale your workload at all.
+
+## Implementation History
+
+- 2025-15-11: initial version
```

Comment:
I know we didn't write anything down for AEP-4016 in terms of graduation criteria, but since we went through the process of graduating that one from alpha to beta, I'm wondering if we should have some sort of idea for this one?

I don't know if we should have some formal process, but just judging from the last graduation, I think it make senses to say we would keep it in alpha for one release cycle to allow early adoption, and if there's no graduation bugs/blockers that come up in the issues, then we are okay to graduate to beta.